api/src/test/java/org/openmrs/module/attachments/AttachmentsServiceTest.java

```diff
@@ -0,0 +1,58 @@
+package org.openmrs.module.attachments;
+
+import org.junit.Test;
+import org.openmrs.Encounter;
+import org.openmrs.Patient;
+import org.openmrs.Visit;
+import org.openmrs.api.EncounterService;
+import org.openmrs.api.PatientService;
+import org.openmrs.api.VisitService;
+import org.openmrs.module.attachments.obs.Attachment;
+import org.openmrs.test.BaseModuleContextSensitiveTest;
+import org.openmrs.test.Verifies;
+import org.openmrs.web.test.BaseModuleWebContextSensitiveTest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class AttachmentsServiceTest extends BaseModuleContextSensitiveTest {
+	
+	@Autowired
+	private AttachmentsService as;
+	
+	@Autowired
+	@Qualifier("patientService")
+	protected PatientService ps;
+	
+	@Autowired
+	@Qualifier("visitService")
+	protected VisitService vs;
+	
+	@Autowired
+	@Qualifier("encounterService")
+	private EncounterService es;
+	
+	@Test
+	@Verifies(value = "the parameters should return the corresponding attachments", method = "getAttachments(patient, visit, encounter, includeRetired)")
+	public void getAttachments_shouldReturnAttachments() {
+		
+		// Setup
```

Comment:
This is currently the best way (in the context of tests) to save a complex obs, which in turns is nothing else than an attachment.
- [This](https://github.com/openmrs/openmrs-module-attachments/blob/645101fddefd82e21af7e6c0db0225c967b42900/api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java#L160-L161) would be an example of how to save an image attachment/file.
- [This](https://github.com/openmrs/openmrs-module-attachments/blob/645101fddefd82e21af7e6c0db0225c967b42900/api/src/test/java/org/openmrs/module/attachments/obs/TestHelper.java#L133-L134) would be an example of how to save any other attachment/file.

I suggest to either just do that in your setup or - even better - extend `TestHelper` so that it can easily help save a bunch of attachments at once. I think the latter is far better.

Could you work on a new method in that class that would just do that. Something _like_
```java
public List<Obs> saveComplexObs(int count) {
  ...
}
```
Once you have those obs at hand it is easy to generate a list of attachments. Your test would verify that the list coming out of your service is indeed the same.

The file bytes content could be randomly generated for each file type as in [here](https://github.com/openmrs/openmrs-module-attachments/blob/645101fddefd82e21af7e6c0db0225c967b42900/omod-1.10/src/test/java/org/openmrs/module/attachments/rest/AttachmentController1_10Test.java#L47).

Reply:
@mks-d Thank you very much for the guidance. I pushed the latest changes. Feel free for your feedback.